media/js/firefox/os/firefox-os-new.js

```diff
@@ -0,0 +1,237 @@
+        // scales and fades the large phone image
+        $phoneViewContainer.toggleClass('scale');
+        // activate the demo area
+        $demoContainer.toggleClass('fxos-active');
+
+        // if a demo was specified show and enable animation
+        if (demo) {
+            demo.addClass('animate');
+            setActiveButton($demoContainer, $(demo.data('button')));
+        }
+    }
+
+    $phoneViewContainer.on('click', 'a', function(event) {
+
+        var targetID = event.currentTarget.id;
+
+        // only handle clicks for the demo triggers
+        if (targetID.indexOf('trigger') > -1) {
+            event.preventDefault();
+            var demo = targetID === 'trigger-swipe' ? $swipeApps : $scrollHomescreen;
+            switchState(demo);
+        }
+    });
+
+    $demoContainer.on('click', 'a', function(event) {
+
+        event.preventDefault();
+        var targetID = event.currentTarget.id;
+
+        if (targetID === 'close-demo') {
+            // close the demo view and switch back to default
+            switchState();
+        } else if (targetID === 'scroll-homescreen') {
+            // show the scroll homescreen animation
+            $swipeApps.removeClass('animate');
+            setActiveButton($demoContainer);
+            $scrollHomescreen.addClass('animate');
+        } else if (targetID === 'swipe-apps') {
+            // show swipe apps anim and make it play
+            $scrollHomescreen.removeClass('animate');
+            setActiveButton($demoContainer);
+            $swipeApps.addClass('animate');
+        }
+    });
+
+    var COUNTRY_CODE = '';
+    /*
+    * Get country code via geo-ip lookup
+    */
+    function getGeoLocation () {
+        try {
+            console.log('COUNTRY_CODE', geoip_country_code());
+            COUNTRY_CODE = geoip_country_code().toLowerCase();
+        } catch (e) {
+            console.log('error', e);
+            COUNTRY_CODE = '';
+        }
+    }
+
+    /*
+    * Set newsletter default options based on cc
+    */
+    function setNewsletterDefaults () {
+        var countrySelection = $('#id_country option[value='+ COUNTRY_CODE +']');
+
+        // if cc is in the country option list, select it
+        // remove the default selection first
+        if (countrySelection.length !== 0) {
+            $('#id_country option:selected').prop('selected', false);
+            countrySelection.prop('selected', true);
+        }
+
+        // for ve and co set the default language to es
+        // other countries auto select based on cc
+        switch(COUNTRY_CODE) {
+        case 've':
+            $('#id_lang option[value="es"]').prop('selected', true);
+            break;
+        case 'co':
+            $('#id_lang option[value="es"]').prop('selected', true);
+            break;
+        default:
+            $('#id_lang option[value="' + COUNTRY_CODE + '"]').prop('selected', true);
+        }
+    }
+
+    window.trackGAEvent = function (eventsArray, callback) {
```

Comment:
This is old GA tracking code and can be replaced. We now use a [global GA helper](https://github.com/mozilla/bedrock/blob/master/media/js/base/global.js#L153) that can be called like so:

    gaTrack(['_trackEvent', '/os/devices/ Interactions', 'foo', 'bar'], callback);